src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/BackpackListener.java

```diff
@@ -105,8 +107,15 @@ private boolean isAllowed(@Nonnull SlimefunBackpack backpack, @Nullable ItemStac
         if (item == null || item.getType() == Material.AIR) {
             return true;
         }
+        SlimefunItem sfitem = SlimefunItem.getByItem(item);
+        if (backpack instanceof HazmatBackpack && !(sfitem instanceof Radioactive)) {
+            return false;
+        }
+        if (sfitem instanceof Radioactive && !(backpack instanceof HazmatBackpack)) {
+            return false;
+        }
```

Comment:
This logic should absolutely NOT happen here...
It should be applied in the `isItemAllowed` method of the respective backpack, non-nuclear for normal ones and also nuclear for hazmat ones.

Reply:
Should I add both logics in SlimefunBackpack?
Or would I have to add each logic for each backpack implementation of it?

Reply:
You should generally keep the log in the respective class.
So normal SlimefunBackpack logic -> SlimefunBackpack.java
HazmatBackpack logic -> HazmatBackpack.java

The HazmatBackpack could just fallback to the super logic if it isn't radioactive, to avoid redundancies.